### ISSUE TITLE
Flaky `tests\tcp::merge`

### ISSUE THREAD

**Issue by ktff** [timberio/vector#3041]
Ref. #2978 

Tests that we got all of the events in the correct order from a `tcp` sink merged from outputs of two `tcp` sources. 

https://github.com/timberio/vector/pull/3010/checks?check_run_id=861595079

**Comment by ktff** [comment]
This is also affected by #3308 and it seams that it's the main cause.

**Comment by binarylogic** [comment]
Thanks @ktff, I assume you'll revive the test?

**Comment by ktff** [comment]
@binarylogic yes.

**Comment by ktff** [comment]
Nope, CI proved me [wrong](https://github.com/timberio/vector/pull/3354/checks?check_run_id=948262780). 

#3308 is affecting it, but there is some other similar race.

**Comment by binarylogic** [comment]
@ktff thanks for trying. Should be table this for now or do you think it's worth investigating the issue?

**Comment by ktff** [comment]
Table it. As mentioned, `merge_and_fork` covers this test, so the issue is probably with the test itself.

**Comment by jszwedko** [comment]
Closing since this test was deleted and evidently doesn't appear to have been needed.
